crates/notion-core/src/tool.rs

```diff
@@ -330,4 +334,17 @@ impl Tool for Yarn {
     fn command(self) -> Command {
         self.0
     }
+
+    /// Perform any tasks which must be run after the tool runs but before exiting.
+    fn finalize(session: &Session, maybe_status: &io::Result<ExitStatus>) {
+        if let Ok(_) = maybe_status {
+            if let Some(project) = session.project() {
+                let errors = project.autoshim();
+
+                for error in errors {
+                    display_error(&error);
```

Comment:
I think I'm starting to doubt the part of the exit codes RFC that says that we should preserve the exit code of the underlying tool even if extra logic afterwards fails. We don't need to block this PR on this, but I [filed an RFC issue](https://github.com/notion-cli/rfcs/issues/26) to follow up.